smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/language/syntax/parameters/ParameterType.java

```diff
@@ -45,6 +51,9 @@ public static ParameterType fromNode(StringNode node) throws RuleError {
         if (value.equalsIgnoreCase("Boolean")) {
             return BOOLEAN;
         }
+        if (value.equalsIgnoreCase("StringArray")) {
+            return STRING_ARRAY;
+        }
         throw new RuleError(new SourceException(
                 String.format("Unexpected parameter type `%s`. Expected `String` or `Boolean`.", value), node));
```

Comment:
>Hmm, should we actually allow case insensitive here? Don't think we do that anywhere else.

This comment probably falls in the category of - this was discussed before in design already (that I'm unaware since new to this features) and we can't change this now anyways. So this part of the comment is non-blocking.

Reply:
I was following the pattern for the other parameter types (see above lines), which are case insensitive.

Reply:
Yeah, continuing with case insensitive is fine. But still missing StringArray in the exception message.

Reply:
Also, is this error message customer facing to the model author? If so, do we want to matching the casing used in the spec?

```suggestion
                String.format("Unexpected parameter type `%s`. Expected `string`, `boolean` or `stringArray`.", value), node));
```

And if so, might be better to change the `equalsIgnoreCase` and `toString()` to also use the same casing in the Java code, so it's the same case across the Java code.

Reply:
Yeah, good call - I'll update to match the casing of the spec everywhere.